src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8662,13 +8626,36 @@ namespace WaterThermalTanks {
         Array1D<Real64> Par(5);     // Parameters passed to RegulaFalsi
         static Real64 MinTemp(0.0); // used for error messages, C
         std::string IterNum;        // Max number of iterations for warning message
+        Real64 TimeElapsed = HourOfDay + TimeStep * TimeStepZone + SysTimeElapsed;
 
         // FLOW:
         DesuperheaterNum = WaterThermalTank(WaterThermalTankNum).DesuperheaterNum;
         AvailSchedule = GetCurrentScheduleValue(WaterHeaterDesuperheater(DesuperheaterNum).AvailSchedPtr);
         WaterInletNode = WaterHeaterDesuperheater(DesuperheaterNum).WaterInletNode;
         WaterOutletNode = WaterHeaterDesuperheater(DesuperheaterNum).WaterOutletNode;
 
+        if (DesuperheaterTimeElapsed != TimeElapsed && WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_AIR_WATER_HEATPUMP_EQ) {
+            DataHeatBalance::HeatReclaimSimple_WAHPCoil(WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSourceIndexNum).DesuperheaterReclaimedHeat = 0;
+            DesuperheaterTimeElapsed = TimeElapsed;
+        }
+
+        // store first iteration tank temperature and desuperheater mode of operation
+        if (FirstHVACIteration && !ShortenTimeStepSys && WaterHeaterDesuperheater(DesuperheaterNum).FirstTimeThroughFlag) {
+            // Save conditions from end of previous system timestep
+            // Every iteration that does not advance time should reset to these values
+            WaterThermalTank(WaterThermalTankNum).SavedTankTemp = WaterThermalTank(WaterThermalTankNum).TankTemp;
+            WaterThermalTank(WaterThermalTankNum).SavedSourceOutletTemp = WaterThermalTank(WaterThermalTankNum).SourceOutletTemp;
+            WaterHeaterDesuperheater(DesuperheaterNum).SaveMode = WaterHeaterDesuperheater(DesuperheaterNum).Mode;
+            WaterHeaterDesuperheater(DesuperheaterNum).FirstTimeThroughFlag = false;
+            if (WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_AIR_WATER_HEATPUMP_EQ){
+                DataHeatBalance::HeatReclaimSimple_WAHPCoil(WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSourceIndexNum).DesuperheaterReclaimedHeat += WaterHeaterDesuperheater(DesuperheaterNum).HeaterRate;
```

Comment:
Save the desuperheater rate here (based on firstHVACiteration signal) caused the reclaimed heat being tracked one time step behind of hvac iteration.